crates/compiler/builtins/bitcode/src/list.zig

```diff
@@ -751,7 +751,7 @@ pub fn listConcat(list_a: RocList, list_b: RocList, alignment: u32, element_widt
             return list_b;
         } else {
             // we must consume this list. Even though it has no elements, it could still have capacity
-            list_b.deinit(usize);
+            utils.decref(list_b.bytes, list_b.capacity, alignment);
```

Comment:
I think we should change list.deinit to take the alignment of the argument and then remove all of these direct calls to utils.decref.

I think it was just a mistake in api to make list.deinit take a type. As such it is essentially not used anyway and we keep falling back on directly calling utils.decref.

Reply:
I changed it back for the original call, but am not sure about other calls to `deinit` that call it with different arguments than `bytes` and `capacity`.

Reply:
If you look at the source of `utils.decref` all that size is used for is to tell if the list has an allocation or not. So capacity should always work. If the size is 0, there is no alloction and `utils.decref` does nothing. So it should be safe to always call `list.deinit(alignment)` as long as the capacity is actually set for the list.

Reply:
I updated the other calls, but replacing `length` with `capacity` does mean we decref lists that were not before (that did have capacity but no items). Whether this is correct, idk.

Reply:
It is very likely correct. We added capacity later in the development of lists and strings. As such, many things did not properly get updated to use capacity. I have a fuzzer and have been slowly adding list and string functions to it. The most common memory leaks are caused by not freeing empty (length == 0) lists/strings that have a capacity.